mapbox/dr-ui
# Replace stateless components with pure components

**This branch will merge into #431** 

This PR updates eslint-config-docs which includes a new rule [1prefer-stateless-function1](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md) and [`prefer-const`](https://eslint.org/docs/rules/prefer-const). Following the guidance from eslint, this PR updates all eligible components to use PureComponent. This PR also updates components with state, where the values are primitives (not objects or arrays).

Ref https://github.com/mapbox/docs-lighthouse/issues/2

## How to test

You could open the test cases app and make sure components that have been updated in the PR render as expected.

## QA checklist


Create beta release and test in repo:
- [x] All Segment data is coming through
- [x] Feedback data comes through

Check all interactive components:
- [x] RelatedPage (video modal)
- [x] PageLayout (example pages)
- [x] CodeToggle



## Before merge
